homeassistant/helpers/selector.py

```diff
@@ -179,8 +179,8 @@ class NumberSelector(Selector):
 
     CONFIG_SCHEMA = vol.Schema(
         {
-            vol.Required("min"): vol.Coerce(float),
-            vol.Required("max"): vol.Coerce(float),
+            vol.Optional("min"): vol.Coerce(float),
+            vol.Optional("max"): vol.Coerce(float),
             vol.Optional("step", default=1): vol.All(
                 vol.Coerce(float), vol.Range(min=1e-3)
             ),
```

Comment:
Should the default step only be set if mode is "slider"?

Reply:
No, steps also work on the input box.

Reply:
OK, but do we want a default step size for the box mode?

Reply:
Our default step size for numbers is 1 for both input_number and number entities. That is where this originates from I guess. I think it is OK?

Reply:
OK, but what does it mean in box mode?
If the user inputs `0.5` in the box, with no step set, the input is rounded to 1?
How can we allow input of an arbitrary number with a minimum step of 0.001?

For the integrations like the threshold integration, we have no idea what the data is, so the user input should just be a number with no constraints I think: https://www.home-assistant.io/integrations/threshold#hysteresis

We can save this for a separate PR since this is about aligning core with new features in with frontend though.

Reply:
I misunderstood what `step` does; it sets slider steps and up/down keyboard binding for the box. The user's input is not restricted regardless of `step`.